library/core/src/slice/mod.rs

```diff
@@ -883,6 +883,36 @@ impl<T> [T] {
         ChunksExactMut::new(self, chunk_size)
     }
 
+    /// Splits the slice into a slice of `N`-element arrays,
+    /// starting at the beginning of the slice,
+    /// and a remainder slice with length strictly less than `N`.
+    ///
+    /// # Panics
+    ///
+    /// Panics if `N` is 0. This check will most probably get changed to a compile time
+    /// error before this method gets stabilized.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// #![feature(slice_as_chunks)]
+    /// let slice = ['l', 'o', 'r', 'e', 'm'];
+    /// let (chunks, remainder) = slice.as_chunks();
```

Comment:
Should the example be: `let (...) = slice.as_chunks::<2>();`?  (Similar for `as_chunks_mut`.)

I imagine this example with implicit N only works because `N` is inferred from the subsequent `assert_eq!` structure.  (But I'm pretty new to Rust, so maybe I'm missing something!)

Thanks!

Reply:
It does indeed infer `N`, and that's meant to be a good thing, but perhaps the example should have a comment pointing this out.

Reply:
We could either put `slice.as_chunks()` as comment or `slice.as_chunks::<2>()` as comment to show that it is being infered.

But in some cases type is not infered which may be confusing, like `Some([])` cannot be infered as `N` for `[T; N]` so it may contradict to this comment here, so it would also be safer to say "most of the time it could be infered" for now. I recall seeing this in one of the pull request.

Reply:
Yes, it does infer `N` in these cases.  I agree it would be nice to also have some examples that turbofish it.

Maybe some like these, to emphasize the length math:
```rust
let slice = ['-'; 100];
let (chunks, remainder) = slice.as_chunks::<20>();
assert_eq!((chunks.len(), remainder.len()), (5, 0));
let (chunks, remainder) = slice.as_chunks::<30>();
assert_eq!((chunks.len(), remainder.len()), (3, 10));
```